### PR TITLE
Update AWS SDKs + Hudi

### PR DESCRIPTION
**Pull request by wendigo**
New major AWS SDK v2 deprecated ADAPTIVE retry strategy and introduces ADAPTIVE_V2 instead.

### CODE REVIEW

**Comment by wendigo** [comment]
@losipiuk for exchange manager changes

File: lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ public static software.amazon.awssdk.core.retry.RetryMode getRetryMode(RetryMode
             return switch (retryMode) {
                 case STANDARD -> software.amazon.awssdk.core.retry.RetryMode.STANDARD;
                 case LEGACY -> software.amazon.awssdk.core.retry.RetryMode.LEGACY;
-                case ADAPTIVE -> software.amazon.awssdk.core.retry.RetryMode.ADAPTIVE;
+                case ADAPTIVE -> software.amazon.awssdk.core.retry.RetryMode.ADAPTIVE_V2;
```
Comment (anusudarsan) [inline lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java:74]: please update file-system-s3 doc

**Review by anusudarsan — APPROVED** [review]
LGMT % one comment to update docs

File: lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java [hunk 2]
```diff
@@ -55,16 +54,13 @@ public final class S3FileSystemFactory
...
-                .numRetries(config.getMaxErrorRetries())
-                .build();
         s3.overrideConfiguration(ClientOverrideConfiguration.builder()
                 .addExecutionInterceptor(AwsSdkTelemetry.builder(openTelemetry)
                         .setCaptureExperimentalSpanAttributes(true)
                         .setRecordIndividualHttpError(true)
                         .build().newExecutionInterceptor())
-                .retryPolicy(retryPolicy)
+                .retryStrategy(getRetryMode(config.getRetryMode()))
+                .retryStrategy(strategy -> strategy.maxAttempts(config.getMaxErrorRetries()))
```
Comment (losipiuk) [inline lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java:63]: .retryStrategy(SdkDefaultRetryStrategy.forRetryMode(getRetryMode(config.getRetryMode())).toBuilder()
                        .maxAttempts(config.getMaxErrorRetries())
                        .build())
```

Your code would not work as second call to `retryStrategy` resets the retry mode.

File: plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/S3FileSystemExchangeStorage.java [hunk 3]
```diff
@@ -158,11 +157,9 @@ public S3FileSystemExchangeStorage(S3FileSystemExchangeStorageStats stats, Excha
         this.compatibilityMode = requireNonNull(compatibilityMode, "compatibilityMode is null");
 
         AwsCredentialsProvider credentialsProvider = createAwsCredentialsProvider(config);
-        RetryPolicy retryPolicy = RetryPolicy.builder(config.getRetryMode())
-                .numRetries(config.getS3MaxErrorRetries())
-                .build();
         ClientOverrideConfiguration overrideConfig = ClientOverrideConfiguration.builder()
-                .retryPolicy(retryPolicy)
+                .retryStrategy(config.getRetryMode())
+                .retryStrategy(builder -> builder.maxAttempts(config.getS3MaxErrorRetries()))
```
Comment (losipiuk) [inline plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/S3FileSystemExchangeStorage.java:162]: fix here too

**Review by losipiuk — CHANGES_REQUESTED** [review]
need fixing

File: lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java (hunk 2 above)
Comment (wendigo) [reply lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java:63]: Nice catch.

**Review by losipiuk — APPROVED** [review]
LGTM % fix needed

File: lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java (hunk 1 above)
Comment (wendigo) [reply lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java:74]: What should I write there except for the fact that ADAPTIVE_V2 is no longer experimental?

**Pull request merged**